### PR TITLE
add hasSelectedInnerBlock hook

### PR DESCRIPTION
**Pull request by fabiankaegy**
This also is something that I needed to use several times on projects. There was no real good way of knowing wether a block has selected inner blocks. And this hook allows you to check that. 

```js
import { useHasSelectedInnerBlock } from '@10up/block-components';
function BlockEdit( props ) {
    const hasSelectedInnerBlock = useHasSelectedInnerBlock(props);
    return (
        <p>This block is currently { hasSelectedInnerBlock ? 'selected' : 'not selected' }.</p>
    )
}

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,10 +37,27 @@ apiFetch( {
 } )...
 ```
 
+## useHasSelectedInnerBlock
+
+Determine wether one of the inner blocks currently is selected.
```
Comment (ryanwelcher) [inline README.md:42]:
```suggestion
Determine whether one of the inner blocks currently is selected.
```

File: hooks/use-has-selected-inner-block.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { useSelect } from '@wordpress/data';
+
+/**
+ * check wether the block to check has any children that match the clientID of the selected block
```
Comment (ryanwelcher) [inline hooks/use-has-selected-inner-block.js:4]:
```suggestion
 * check whether the block to check has any children that match the clientID of the selected block
```

File: hooks/use-has-selected-inner-block.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @param {WPBlock} selectedBlock Block object of the currently selected block
+ * @param {string} clientId Client ID of the block to check
+ * @return {boolean} Wether or not a child block is selected
```
Comment (ryanwelcher) [inline hooks/use-has-selected-inner-block.js:8]:
```suggestion
 * @return {boolean} Whether or not a child block is selected
```

File: hooks/use-has-selected-inner-block.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+			isChildSelected = hasSelectedInnerBlock(selectedBlock, innerBlock.clientId);
+		}
+	});
+
+	return isChildSelected;
+}
+
+/**
+ * useHasSelectedInnerBlock
+ * Determine wether one of the inner blocks currently is selected
```
Comment (ryanwelcher) [inline hooks/use-has-selected-inner-block.js:32]:
```suggestion
 * Determine whether one of the inner blocks currently is selected
```

File: hooks/use-has-selected-inner-block.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @param {WPBlock} block Block object to with children
+ * @return {boolean} wether or not any children are selected.
```
Comment (ryanwelcher) [inline hooks/use-has-selected-inner-block.js:35]:
```suggestion
 * @return {boolean} whether or not any children are selected.
```

**Review by ryanwelcher — CHANGES_REQUESTED** [review]
@fabiankaegy this is great! I had a HOC version of this from a past project but  having a hook is much better. I added some suggestions for spelling but they are just nitpicks. I think this is OK to merge once updated.

**Comment by fabiankaegy** [comment]
@ryanwelcher I updated the suggestions :) Merging now  👍

**Pull request merged**